src/lib/Sympa/Message.pm

```diff
@@ -3700,6 +3708,26 @@ sub get_id {
     return join '/', grep {$_} ($id, $shelved);
 }
 
+# autodetect all URIs in block of text and wrap them in HTML tags
+sub uri_finder {
+    my $text = shift;
+
+    # insert "mailto:" on all email addresses, so URI::Find will find them:
+    $text =~ s,(^|\s)(\S+)@(\S+)($|\s),$1mailto:$2\@$3$4,g;
+
+    # wrap all URIs in HTML tags
+    my $finder = URI::Find::Schemeless->new(sub {
+        my($uri, $orig_uri) = @_;
+        return qq|<a href="$uri">$orig_uri</a>|;
+                                            });
+    $finder->find(\$text, \&CGI::escapeHTML);
+
+    # and finally clean up the displayed link text on mailto links:
+    $text =~ s/>mailto:/>/g;
```

Comment:
I personally think that this “`mailto:`” is included in the original text and may not be removed.

Reply:
Please feel free to submit a new PR until you're satisfied!

I’ll play with this PR this weekend.